src/main/java/com/palantir/docker/compose/execution/DockerCompose.java

```diff
@@ -54,6 +55,11 @@ public DockerCompose(DockerComposeExecutable rawExecutable, DockerMachine docker
         this.dockerMachine = dockerMachine;
     }
 
+    public DockerCompose withAdditionalComposeFile(File composeFile) {
```

Comment:
I'm not sure this is the right place to put it?

Surely by the time that you have created your `DockerComposition` you would already know the files that you need. It feels like this class will then turn into something that's somewhat a builder but isn't. 

I think action I'm suggesting is: if you really have to, make it a builder, but on the contrary, I don't think it needs to live here since `DockerCompose` isn't public API essentially, and everything should be known by the time a `DockerComposition` is built using its builder.